beater/pub.go

```diff
@@ -85,14 +90,22 @@ func (p *publisher) Stop() {
 // Send tries to forward pendingReq to the publishers worker. If the queue is full,
 // an error is returned.
 // Calling send after Stop will return an error.
-func (p *publisher) Send(req pendingReq) error {
+func (p *publisher) Send(ctx context.Context, req pendingReq) error {
 	p.m.RLock()
 	defer p.m.RUnlock()
 	if p.stopped {
 		return errChannelClosed
 	}
 
+	span, ctx := elasticapm.StartSpan(ctx, "Send", "Publisher")
+	if span != nil {
+		defer span.Done(-1)
+		req.trace = true
+	}
+
 	select {
+	case <-ctx.Done():
+		return ctx.Err()
```

Comment:
could be the case that if the agent fails to instrument a request, for whatever reason, we just balk out and do not process it?

Reply:
Nope. Already in some cases, ctx doesn't contain a transaction or span, namely when the request is coming through the pipe listener.